package/MDAnalysis/lib/formats/libdcd.pyx

```diff
@@ -0,0 +1,403 @@
+        self.fname = fname.encode('utf-8')
+        self.n_atoms = 0
+        self.is_open = False
+        if path.isfile(self.fname) and mode == 'w':
+            raise IOError('''Aborting -- attempted to overwrite an
+                                existing file path.''')
+        self.open(self.fname, mode)
+
+    def __dealloc__(self):
+        self.close()
+
+    def __enter__(self):
+        """Support context manager"""
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        """Support context manager"""
+        self.close()
+        # always propagate exceptions forward
+        return False
+
+    def __iter__(self):
+        self.close()
+        self.open(self.fname, self.mode)
+        return self
+
+    def __next__(self):
+        if self.reached_eof:
+            raise StopIteration
+        return self.read()
+
+    def __len__(self):
+        if not self.is_open:
+            raise IOError('No file currently opened')
+        return self.n_frames
+
+    def tell(self):
+        return self.current_frame
+
+    def open(self, filename, mode='r'):
+        if self.is_open:
+            self.close()
+
+        if mode == 'r':
+            fio_mode = FIO_READ
+        elif mode == 'w':
+            fio_mode = FIO_WRITE
+        else:
+            raise IOError("unkown mode '{}', use either r or w".format(mode))
+        self.mode = mode
+
+        ok = fio_open(self.fname, fio_mode, <fio_fd*> &self.fp)
+        if ok != 0:
+            raise IOError("couldn't open file: {}\n"
+                          "ErrorCode: {}".format(self.fname, ok))
+        self.is_open = True
+        self.current_frame = 0
+        if self.mode == 'r':
+            self.remarks = self._read_header()
+        self.reached_eof = False
+
+    def close(self):
+        if self.is_open:
+            # In case there are fixed atoms we should free the memory again.
+            # Both pointers are guaranted to be non NULL if either one is.
+            if self.freeind != NULL:
+                close_dcd_read(self.freeind, self.fixedcoords);
+
+            ok = fio_fclose(self.fp)
+            self.is_open = False
+            if ok != 0:
+                raise IOError("couldn't close file: {}\n"
+                            "ErrorCode: {}".format(self.fname, ok))
+
+
+    def _read_header(self):
+        if not self.is_open:
+            raise IOError("No file open")
+
+        cdef char* c_remarks
+        cdef int len_remarks = 0
+
+        ok = read_dcdheader(self.fp, &self.n_atoms, &self.nsets, &self.istart,
+                            &self.nsavc, &self.delta, &self.nfixed, &self.freeind,
+                            &self.fixedcoords, &self.reverse_endian,
+                            &self.charmm, &c_remarks, &len_remarks)
+        if ok != 0:
+            raise IOError("Reading DCD header failed: {}".format(DCD_ERRORS[ok]))
+
+        if c_remarks != NULL:
+            py_remarks = <bytes> c_remarks[:len_remarks]
```

Comment:
Should we maybe do some parsing of the bytes we receive as well when reading the header?